stl/inc/format

```diff
@@ -3286,10 +3286,15 @@ struct _Formatter_base {
                 _FormatCtx.arg(static_cast<size_t>(_Specs._Dynamic_precision_index)));
         }
 
-        return _STD visit_format_arg(
-            _Arg_formatter<typename _FormatContext::iterator, _CharT>{
-                ._Ctx = _STD addressof(_FormatCtx), ._Specs = _STD addressof(_Format_specs)},
-            basic_format_arg<_FormatContext>{_Val});
+        _Arg_formatter<typename _FormatContext::iterator, _CharT> _Arg_fmt{
+            ._Ctx = _STD addressof(_FormatCtx), ._Specs = _STD addressof(_Format_specs)};
+
+        using _Storage_type = typename _Format_arg_traits<_FormatContext>::template _Storage_type<_Ty>;
+        if constexpr (is_same_v<typename basic_format_arg<_FormatContext>::handle, _Storage_type>) {
+            return _STD visit_format_arg(_Arg_fmt, basic_format_arg<_FormatContext>{_Val});
+        } else {
+            return _STD visit_format_arg(_Arg_fmt, basic_format_arg<_FormatContext>{static_cast<_Storage_type>(_Val)});
+        }

```

Comment:
Partially informed drive-by comment: I would _much_ prefer that we correctly implement `basic_format_arg`s constructor overload set instead of adding external hacks like this.

Reply:
Ah, I see! I didn't realize how messed up `basic_format_arg`'s constructors were in the Standardese.

I agree with your suggested approach - handle `long` and `unsigned long` (I think those are the affected arithmetic types?), fix the modifiable `basic_string` issue, and fix the (apparent) `charT *` issue all in the overload set. This seems like an acceptable reason to (temporarily) diverge from the Standardese, under the "clearly broken wording" exception.